pkg/utils/utils.go

```diff
@@ -0,0 +1,20 @@
+package utils
+
+import (
+	"crypto/sha512"
+	"fmt"
+)
+
+// 29 - len('CNI') - 2*len('-')
+const maxNameLen = 16
+
+// Generates a chain name to be used with iptables.
+// Ensures that the generated name is less than
+// 29 chars in length
+func FormatChainName(name string, id string) string {
+	h := sha512.Sum512([]byte(id))
+	if len(name) > maxNameLen {
+		return fmt.Sprintf("CNI-%s-%x", name[:len(name)-maxNameLen], h[:8])
```

Comment:
Not sure to understand... shouldn't it be `name[:maxNameLen]`?

Also, to distinguish "verylongnameofthenetwork01" and "verylongnameofthenetwork02" (both truncated to "verylongnameofth"), could the hash `h` be feed with both `name` and `id`?